src/handshakes/abstract-handshake.ts

```diff
@@ -104,10 +106,7 @@ export abstract class AbstractHandshake {
 
   protected dh(privateKey: bytes32, publicKey: bytes32): bytes32 {
     try {
-      const derived = x25519.derive(publicKey, privateKey);
-      const result = Buffer.alloc(32);
-      derived.copy(result);
-      return result;
+      return Buffer.from(box.before(publicKey, privateKey), 0, 32);
```

Comment:
What are you trying to achieve here?
Buffer view over 0...32 byte of output?

Reply:
I unnecessarily tried to replicate the old logic.